src/pages/index.js

```diff
@@ -20,88 +87,176 @@ const cardSection = new Section(
   ".gallery__cards"
 );
 
-// Render initial cards
-cardSection.renderItems();
+function fillCardLike(){
+  const likeBtn = document.querySelector('.card__like-button');
+  likeBtn.classList.toggle('card__like-button--active');
+}
 
-//---------------------FORMS AND SUBMITS--------------------------
+function emptyCardLike(){
+  const likeBtn = document.querySelector('.card__like-button');
+  likeBtn.classList.remove('card__like-button--active');
+}
+
+// Fetch API cards and render them inside the same section
+api.getInitialCards()
+.then((cards) => {
+  cards.forEach((cardData) => {
+    const cardElement = createCard(cardData); // Generate card using createCard
+    cardSection.addItem(cardElement); // Properly add to `.gallery__cards`
+    
+    console.log(cardData.isLiked)
+    if(cardData.isLiked){
+      console.log(cardData._id);  
+      fillCardLike();
+    }else{
+      emptyCardLike();
+    }
+  });
+});
+
+//---------------------FORMS AND VALIDATION AND SUBMIT BUTTONS RESPONSES--------------------------
 
 // Initialize form validation
 const addCardValidator = new FormValidator(config, document.querySelector("#add__modal_form"));
 const profileValidator = new FormValidator(config, document.querySelector("#edit__modal_form"));
+const avatarValidator = new FormValidator(config, document.querySelector("#avatar__modal_form"));
 addCardValidator.enableValidation();
 profileValidator.enableValidation();
+avatarValidator.enableValidation();
 
 //-----------------------POPUPS--------------------------------
-
 // Create Popups
 const imagePopup = new PopupWithImage("#modal__picture");
 imagePopup.setEventListeners();
 
 //  Initialize `UserInfo`
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
-  jobSelector: ".profile__description"
+  jobSelector: ".profile__description",
+  avatarSelector: ".profile__avatar"
 });
 
 const profilePopup = new PopupWithForm("#profileModal", (formData) => {
-  userInfo.setUserInfo(formData);
+  userInfo.setUserInfo(formData); // Make changes on the page right away
```

Comment:
`userInfo.setUserInfo(formData); `

You need to update the `DOM` only in blocks  `.then` after a successful response from the server, so that when there is an error, it shouldn’t change